pydatastructs/linear_data_structures/tests/test_algorithms.py

```diff
@@ -322,3 +322,24 @@ def test_next_prev_permutation():
         _, prev_array = prev_permutation(array)
         _, orig_array = next_permutation(prev_array)
         assert str(orig_array) == str(array)
+
+def test_linear_search():
+    ODA = OneDimensionalArray
+
+    array = ODA(int, [1, 2, 5, 7, 10, 29, 40])
+    for i in range(len(array)):
+        assert i == linear_search(array, array[i])
+
+def test_binary_search():
+    ODA = OneDimensionalArray
+
+    array = ODA(int, [1, 2, 5, 7, 10, 29, 40])
+    for i in range(len(array)):
+        assert i == binary_search(array, array[i])
+
+def test_jump_search():
+    ODA = OneDimensionalArray
+
+    array = ODA(int, [1, 2, 5, 7, 10, 29, 40])
+    for i in range(len(array)):
+        assert i == jump_search(array, array[i])
```

Comment:
What about corner cases like, empty array, one element array? And see `_test_common_sort` above. Repetition of tests can be avoided by doing something similar here.

Reply:
Are you asking for the test function of the actual algorithm written?

Reply:
Yes. You are using the same test for every search algorithm. So why not write a common test function, `_test_common_search(search_function_api)` and then in `test_jump_search` (for example),

```python
def test_jump_search():
    _test_common_search(jump_search)
```

Reply:
Right, I will do this